### ISSUE TITLE
How do I dynamically create and pass a slice of fieldOpts to GenerateModel function

### ISSUE THREAD

**Issue by sudopower** [go-gorm/gen#369]
## The document you expected this should be explained
https://github.com/go-gorm/gen#generate-model

## Expected answer
A way to import and create a slice of type FieldOpt 

## Use case
I have two tables teams and players. A team can have multiple players 
I need a slice of type players inside type team, which I accomplish using below code

```
#current implementaton
g.GenerateModelAs(table, alias,   
   gen.FieldGORMTag("created_at", "->"),                                  # readonly in CRUD
   gen.FieldNewTag("created_at", "swaggerignore:\"true\""),               # swagger should ignore it
   gen.FieldGORMTag("updated_at", "->"),                                  # readonly in CRUD
   gen.FieldNewTag("updated_at", "swaggerignore:\"true\""),               # swagger should ignore it
   gen.FieldRelateModel(field.HasMany, "Players", models.Players{},       # create a slice of players associated with team
       &field.RelateConfig{
      GORMTag: "foreignKey:TeamID",
  }),
)
```

I'm passing 5 different field options in this example. 
I want to pass it dynamically such that I can create a slice of fieldOpts with FieldGORMTag/FieldNewTag/FieldRelateModel 
outside of GenerateModel function and pass this slice as argument.
Like below
```
#expected implementation
var options []gen.FieldOpt
options = append(options, gen.FieldGORMTag("created_at", "->"))
...
...
options = append(options, gen.FieldRelateModel(field.HasMany, "Players", models.Players{}.......
```

## Issue
FieldOpt type is not exported by gorm-gen 
Hence I cannot implement above logic

## Tl;dr
Can you suggest some solution to pass FieldOpt dynamically to the GenerateModel function or should I create a feature request ?
I'm open to suggestions / alternate implementation.

**Comment by riverchu** [comment]
i refactor FieldOpt to be exported type, u may execute `go get gorm.io/gen@c276bf787fb0db19319506fd4d8fcd38b82847cd` to try this test version

**Comment by sudopower** [comment]
Awesome! 
Thanks :)
